docs/asyncapi.yaml

```diff
@@ -0,0 +1,158 @@
+components:
+  messages:
+    Organisation:
+      name: Organisation
+      contentType: application/json
+      payload:
+        $ref: '#/components/schemas/Organisation'
+    Calculation:
+      name: Calculation
+      contentType: application/json
+      payload:
+        $ref: '#/components/schemas/Calculation'
+  schemas:
+    Organisation:
+      type: object
+      properties:
+        name:
+          $ref: '#/components/schemas/BusinessName'
+        sbi:
+          $ref: '#/components/schemas/SBI'
+        frn:
+          $ref: '#/components/schemas/FRN'
+        addressLine1:
+          $ref: '#/components/schemas/AddressLine1'
+        addressLine2:
+          $ref: '#/components/schemas/AddressLine2'
+        addressLine3:
+          $ref: '#/components/schemas/AddressLine3'
+        city:
+          $ref: '#/components/schemas/City'
+        county:
+          $ref: '#/components/schemas/County'
+        postcode:
+          $ref: '#/components/schemas/Postcode'
+        emailAddress:
+          $ref: '#/components/schemas/Email'
+        updated:
+          $ref: '#/components/schemas/Updated'
+    Calculation:
+      type: object
+      properties:
+        calculationId:
+          $ref: '#/components/schemas/CalculationId'
+        sbi:
+          $ref: '#/components/schemas/SBI'
+        frn:
+          $ref: '#/components/schemas/FRN'
+        calculationDate:
+          $ref: '#/components/schemas/CalculationDate'
+        invoiceNumber:
+          $ref: '#/components/schemas/InvoiceNumber'
+        scheme:
+          $ref: '#/components/schemas/Scheme'
+        updated:
+          $ref: '#/components/schemas/Updated'
+        fundings:
+          type: array
+          items:
+            $ref: '#/components/schemas/Funding'
+    BusinessName:
+      type: string
+      description: The business name
+      example: 'FFC Ltd'
+    SBI:
+      type: number
+      description: Single Business Identifier
+      minimum: 105000000
+      maximum: 999999999
+      example: 123456789
+    FRN:
+      type: number
+      description: Firm Reference Number
+      minimum: 1000000000
+      maximum: 9999999999
+      example: 1234567890
+    Email:
+      type: string
+      description: The business email address
+      example: farm@farms.com
+    AddressLine1:
+      type: string
+      description: The first line of the address
+      example: 1 The Street
+    AddressLine2:
+      type: string
+      description: The second line of the address
+      example: The Area
+    AddressLine3:
+      type: string
+      description: The third line of the address
+      example: The Town
+    City:
+      type: string
+      description: The fourth line of the address
+      example: The Region
+    County:
+      type: string
+      description: The fifth line of the address
+      example: The County
+    Postcode:
+      type: string
+      description: The postcode
+      example: AB1 2CD
+    Updated:
+      type: dateTime
```

Comment:
Not an accepted asyncapi type.

We've used `string` for dates.

https://github.com/DEFRA/ffc-pay-enrichment/blob/main/docs/asyncapi.yaml#L278